aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -531,6 +545,16 @@ func flattenProcessingConfiguration(pc *firehose.ProcessingConfiguration, roleAr
 		return []map[string]interface{}{}
 	}
 
+	enabled := aws.BoolValue(pc.Enabled)
+
+	// The AWS SDK can represent "no processing configuration" in two ways:
+	// 1. With a nil value
+	// 2. With an empty processor list and enabled set to false.
+	// These are equivalent so we use the same state description, to avoid diffs.
+	if !enabled && len(pc.Processors) == 0 {
```

Comment:
Typically, if the configuration block contains a nested "enabled" or similar attribute that we appropriately handle the default case, we prefer to actually pass through the API response into the state as-is (e.g. 1 list item and enabled = false) and then introduce `DiffSuppressFunc: suppressMissingOptionalConfigurationBlock,` on the configuration block attribute. This means the Terraform resource can accept a configuration with `enabled = false` and no processors without showing a difference, while also ignoring the configuration block if all the underlying attribute defaults match. 

This change was good to revert back to how it was. If for some reason this does need to be changed, we should add a covering acceptance test to show the errant behavior in a future change request. 👍